jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java

```diff
@@ -118,4 +124,154 @@ public String describeQueryResultColumns(final String query) throws SQLException
+        if (this.properties.containsKey(property) //
+                && !BOOLEAN_PROPERTY_VALUE_PATTERN.matcher(this.properties.get(property)).matches()) {
+            throw new PropertyValidationException("The value '" + this.properties.get(property) + "' for the property "
+                    + property + " is invalid. It has to be either 'true' or 'false' (case insensitive).");
+        }
+    }
+
+    private void validateDebugOutputAddress() throws PropertyValidationException {
+        if (this.properties.containsKey(DEBUG_ADDRESS_PROPERTY)) {
+            final String debugAddress = this.properties.getDebugAddress();
+            if (!debugAddress.isEmpty()) {
+                final String error = "You specified an invalid hostname and port where a log receiver (e.g. `netcat`) "
+                        + "is listening for incoming connections. The value of the property " + DEBUG_ADDRESS_PROPERTY
+                        + "must adhere to the following format: <host>:<port>, where host is a host name or IP address.";
+                if (debugAddress.split(":").length != 2) {
+                    throw new PropertyValidationException(error);
+                }
+                try {
+                    Integer.parseInt(debugAddress.split(":")[1]);
+                } catch (final NumberFormatException ex) {
+                    throw new PropertyValidationException(error);
+                }
+            }
+        }
+    }
+
+    private void validateExceptionHandling() throws PropertyValidationException {
+        if (this.properties.containsKey(EXCEPTION_HANDLING_PROPERTY)) {
+            final String exceptionHandling = this.properties.getExceptionHandling();
+            if (!((exceptionHandling == null) || exceptionHandling.isEmpty())) {
+                for (final AbstractSqlDialect.ExceptionHandlingMode mode : AbstractSqlDialect.ExceptionHandlingMode
+                        .values()) {
+                    if (!mode.name().equals(exceptionHandling)) {
+                        throw new PropertyValidationException(
+                                "Invalid value '" + exceptionHandling + "' for property " + EXCEPTION_HANDLING_PROPERTY
+                                        + ". Choose one of: " + ExceptionHandlingMode.IGNORE_INVALID_VIEWS.name() + ", "
+                                        + ExceptionHandlingMode.NONE.name());
+                    }
+                }
+            }
+        }
+    }
+
+    protected void validateDialectName(final String dialectName) throws PropertyValidationException {
+        final String availableDialects = "Available dialects: " + SqlDialectRegistry.getInstance().getDialectsString();
+        checkIfContainsDialectName(availableDialects);
+        checkIfDialectIsSupported(availableDialects);
+        checkIfNameIsConsistent(dialectName);
+    }
+
+    private void checkIfNameIsConsistent(final String dialectName) throws PropertyValidationException {
```

Comment:
Reorder `check...` methods in the order they appear above.

Reply:
done